torch/csrc/jit/script/sugared_value.cpp

```diff
@@ -89,6 +89,19 @@ std::shared_ptr<SugaredValue> SimpleValue::attr(
     }
   }
 
+  if (value_->type()->isSubtypeOf(DeviceObjType::get())) {
+    // if (value_->type()->kind() == TypeKind::DeviceObjType) {
+    // property lookups on Device
+    static const std::unordered_set<std::string> fields = {
```

Comment:
Can you refactor this with the tensor fields above ? Maybe something like:
`static const std::unordered_map<TypeKind, std::unordered_set<std::string>> builtin_type_properties`